internal/driver/driver.go

```diff
@@ -2,354 +2,160 @@
 //
 // Copyright (C) 2018 Canonical Ltd
 // Copyright (C) 2018 IOTech Ltd
+// Copyright (C) 2021 Schneider Electric
 //
 // SPDX-License-Identifier: Apache-2.0
 
-//
 package driver
 
 import (
 	"context"
 	"fmt"
-	"log"
 	"sync"
-	"time"
 
-	"github.com/edgexfoundry/device-sdk-go"
-	sdkModel "github.com/edgexfoundry/device-sdk-go/pkg/models"
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
-	"github.com/edgexfoundry/go-mod-core-contracts/models"
-	"github.com/gopcua/opcua"
-	"github.com/gopcua/opcua/ua"
-	"github.com/spf13/cast"
+	"github.com/edgexfoundry/device-opcua-go/internal/config"
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	"github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
 )
 
 var once sync.Once
 var driver *Driver
 
+type safeResourceMap struct {
+	res map[uint32]string
+	mu  sync.Mutex
+}
+
+// Driver struct
 type Driver struct {
 	Logger           logger.LoggingClient
 	AsyncCh          chan<- *sdkModel.AsyncValues
 	CommandResponses sync.Map
-	Config           *configuration
+	serviceConfig    *config.ServiceConfig
+	resourceMap      safeResourceMap
```

Comment:
you can just add `*sync.Mutex` to driver struct
if you want to use `safeResourceMap`, use pointer `*safeResourceMap` here